vidpub/__main__.py

```diff
@@ -96,81 +94,209 @@ def get_match_ratio(session: Session, path: pathlib.Path) -> float:
     return fuzzywuzzy.fuzz.ratio(session.title, path.stem)
 
 
-def choose_video(session: Session) -> pathlib.Path:
+def choose_video(session: Session, video_paths: list) -> pathlib.Path:
     """Look through the file list and choose the one that "looks most like it".
     """
-    score, match = max((get_match_ratio(session, p), p) for p in VIDEO_PATHS)
+    score, match = max((get_match_ratio(session, p), p) for p in video_paths)
     if score < 70:
         raise ValueError("no match")
     return match
 
 
-def build_youtube():
+def build_youtube(action: str):
     from apiclient.discovery import build
     from google_auth_oauthlib.flow import InstalledAppFlow
 
-    flow = InstalledAppFlow.from_client_secrets_file(
-        os.environ["OAUTH2_CLIENT_SECRET"], scopes=[YOUTUBE_UPLOAD_SCOPE]
-    )
-    credentials = flow.run_console()
-    return build("youtube", "v3", credentials=credentials)
+    if action == 'upload':
+        flow = InstalledAppFlow.from_client_secrets_file(
+            os.environ["OAUTH2_CLIENT_SECRET"], scopes=[YOUTUBE_UPLOAD_SCOPE]
+        )
+        credentials = flow.run_console()
+        return build("youtube", "v3", credentials=credentials)
+
+    elif action == 'generate':
+        return build("youtube", "v3", developerKey=YOUTUBE_API_KEY)
```

Comment:
`generate` and `upload` do very different things and should be two different functions. This is bad design.

Reply:
I've updated the code based on the suggestion of separating two different actions into two functions.